### PR TITLE
Add tableName setter for JdbcHttpSessionConfiguration

### PR DESCRIPTION
**Pull request by eddumelendez**
- [X] I have signed the CLA

See gh-488

### CODE REVIEW

**Comment by vpavic** [comment]
@eddumelendez this does not address the #488 since you've only added the setter for ```tableName```.

Also the test you added doesn't actually test anything relevant here. You're just using ```tableName``` setter directly on ```JdbcOperationsSessionRepository``` instead of the one you added in ```JdbcHttpSessionConfiguration```.

**Comment by snicoll** [comment]
@eddumelendez a good way to test this would be to reuse what I've done in Spring Boot (i.e. [extending the regular configuration](https://github.com/snicoll/spring-boot/commit/f46d83938b5668de3d1eec8b0665eebab0edada1#diff-09c1399b224b0c1e813969745a9b661dR39))

**Comment by eddumelendez** [comment]
@vpavic I noticed that when I was looking for Mongo implementation.

@snicoll Thanks for the hint. I will polish this PR.

File: spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java [hunk 1]
```diff
@@ -181,6 +196,17 @@ public PlatformTransactionManager transactionManager() {
 	}
 
 	@Configuration
+	static class CustomTableNameSetConfiguration extends JdbcHttpSessionConfiguration {
+
+		@Autowired
+		public void customize() {
+			DirectFieldAccessor fieldAccessor = new DirectFieldAccessor(this);
+			fieldAccessor.setPropertyValue("tableName", "custom_session");
```
Comment (snicoll) [inline spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java]: uh? Shouldn't you just be using `setTableName`?

**Comment by snicoll** [comment]
It's also missing a `setMaxInactiveIntervalInSeconds`

File: spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java (hunk 1 above)
Comment (eddumelendez) [inline spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java]: my mistake. now it is fixed. thanks!

File: spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java [hunk 2]
```diff
@@ -181,6 +207,17 @@ public PlatformTransactionManager transactionManager() {
 	}
 
 	@Configuration
+	static class CustomTableNameSetConfiguration extends JdbcHttpSessionConfiguration {
+
+		@Autowired
+		public void customize() {
```
Comment (vpavic) [inline spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java]: Why a random ```@Autowired``` method?
Simply overriding the default constructor and using the desired setter inside it is much cleaner and more understandable.

Also separate config for each test are preferred.

Comment (snicoll) [inline spring-session/src/test/java/org/springframework/session/jdbc/config/annotation/web/http/JdbcHttpSessionConfigurationTests.java]: I guess it's taken from what I've done in Spring Boot. I agree an `@Autowired` method with no parameter is not really a good idea.

**Comment by vpavic** [comment]
@eddumelendez Looks good now, thanks for updating the PR.

**Comment by eddumelendez** [comment]
@vpavic thanks to you for the feedback.

**Comment by rwinch** [comment]
Thanks for the PR! This is now merged into master

**Pull request merged**